source/components/events/evregion.c

```diff
@@ -1015,7 +1016,7 @@ AcpiEvOrphanEcRegMethod (
      * with other space IDs to be present; but the code below will then
      * execute the _REG method with the EmbeddedControl SpaceID argument.
      */
-    NextNode = AcpiNsGetNextNode (EcDeviceNode, NULL);
+    NextNode = AcpiNsGetNextNode (DeviceNode, NULL);
     while (NextNode)
     {
         if ((NextNode->Type == ACPI_TYPE_REGION) &&
```

Comment:
Thank you for porting this from the Linux kernel ACPICA to upstream ACPICA.

Unfortunately you missed a bit here.

The `NextNode->Object->Region.SpaceId == ACPI_ADR_SPACE_EC` check needs to be changed to `NextNode->Object->Region.SpaceId == SpaceId` here.